scripts/generateCliCode.ts

```diff
@@ -0,0 +1,348 @@
+      throw new Error('ejs.renderFile returned nothing');
+    }
+    if (outputFile === '-') {
+      console.log(renderedContent);
+
+      return;
+    }
+
+    const parent = path.dirname(outputFile);
+
+    try {
+      fs.accessSync(parent, fs.constants.W_OK | fs.constants.X_OK);
+    } catch (ex: any) {
+      if (ex.code === 'ENOENT') {
+        console.log(`Creating directory ${ parent }...`);
+        fs.mkdirSync(parent, { recursive: true });
+      } else {
+        throw ex;
+      }
+    }
+    fs.writeFileSync(outputFile, renderedContent);
+  }
+
+  protected collectServerSettingsForJSON(settingsTree: Record<string, any>, includeJSONTag: boolean, indent: string): string[] {
+    return Object.keys(settingsTree).map((propertyName) => {
+      return this.collectServerSettingsForJSONProperty(propertyName, settingsTree[propertyName], includeJSONTag, indent);
+    }).flat();
+  }
+
+  protected collectServerSettingsForJSONProperty(propertyName: string, settingsTree: Record<string, any>, includeJSONTag: boolean, indent: string): string[] {
+    const decln = `${ indent }${ propertyName } `;
+
+    if (typeof (settingsTree.type) === 'object') {
+      const lines: string[] = [];
+
+      lines.push(`${ decln } struct {`);
+      lines.push(...this.collectServerSettingsForJSON(settingsTree.type, includeJSONTag, `${ indent }  `));
+      const parts = [indent, '}'];
+
+      if (includeJSONTag) {
+        parts.push(` \`json:"${ uncapitalize(propertyName) }"\``);
+      }
+      lines.push(parts.join(''));
+
+      return lines;
+    }
+    const parts = [decln, ' '];
+
+    if (includeJSONTag) {
+      parts.push('*');
+    }
+    parts.push(settingsTree.type);
+    if (includeJSONTag) {
+      parts.push(`\`json:"${ uncapitalize(propertyName) },omitempty"\``);
+    }
+
+    return [parts.join('')];
+  }
+
+  protected convertStringsToGolang(enums: string[] | undefined): string {
+    return !enums ? '' : `[]string{${ enums.map(s => JSON.stringify(s) ).join(', ') }}`;
+  }
+
+  protected getCommandLineArgValue(lcTypeName: string, capitalizedName: string) {
+    switch (lcTypeName) {
+    case 'bool':
+      return `+"="+strconv.FormatBool(SpecifiedSettings.${ capitalizedName })`;
+    case 'int':
+      return `, strconv.Itoa(SpecifiedSettings.${ capitalizedName })`;
+    case 'string':
+      return `, SpecifiedSettings.${ capitalizedName }`;
+    default:
+      throw new Error(`Unexpected lower-case type name of ${ lcTypeName }`);
+    }
+  }
+
+  protected getFullUsageNote(usageNote: string, aliasFor: string, rawEnums: undefined|string[]): string {
+    const usageParts = [usageNote];
+
+    if (rawEnums) {
+      usageParts.push(`(Allowed values: [${ rawEnums.join(', ' ) }].)`);
+    }
+    if (aliasFor) {
+      usageParts.push(`(Alias for --${ aliasFor }.)`);
+    }
+
+    return usageParts.join(' ').trim();
+  }
+
+  protected updateLeaf(propertyName: string, capitalizedName: string,
+    lcTypeName: string, flagType: string,
```

Comment:
Do we need to pass in a separate `flagType`? It seems to always be `capitalize(lcTypeName)`.

Reply:
Well, right now there's a 1:1 correspondence between the Go type names we use (like 'bool' and 'int') and the names of the methods we call (like `cmd.Flags().BoolVar()` and `cmd.Flags().IntVar()`

I'd rather not count on always having that correspondence, and continue to pass in both the lower-case and capitalized forms of the names.